tap_github/repository_streams.py

```diff
@@ -1916,6 +1920,911 @@ def query(self) -> str:
     ).to_dict()
 
 
+class DiscussionCategoriesStream(GitHubGraphqlStream):
+    """
+    Defines stream fetching discussions categories from each repository.
+
+    This stream is full drop. Categories are returned alphabetically
+    by GitHub, not chronologically. This means the smart pagination and
+    resuming if interrupted features are not supported.
+    """
+
+    name = "discussion_categories"
+    query_jsonpath = "$.data.repository.discussionCategories.nodes.[*]"
+    primary_keys: ClassVar[list[str]] = [
+        "node_id"
+    ]  # id:databaseId is not available for the categories object
+    replication_method = "full_table"
+    parent_stream_type = RepositoryStream  # Github allows a maximum of 25 categories per repository. # noqa: E501
+    ignore_parent_replication_key = True  # Repository's updated_at does not change when a new discussion category is added/modified  # noqa: E501
+
+    def get_records(self, context: Context | None = None) -> Iterable[dict[str, Any]]:
+        """
+        Return a generator of row-type dictionary objects.
+        If discussions are not enabled, skip the API call.
+        """
+        repo = context.get("repo", "unknown")
+        org = context.get("org", "unknown")
+        if not context.get("has_discussions", False):
```

Comment:
Documenting context: This is a nice way to cut down calls in a context where there are lots of repos but only a small fraction are using discussions.